### PR TITLE
goeun / edges / api muncher 

### PR DESCRIPTION
**Pull request by goeunpark**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, and how did you try querying the API? | used postman, messed around with data structures in pry 
What is an API Wrapper? Why is it in `lib`? How would your project change if you needed to interact with more than one API (aka more than just the Edamam API)? |
Describe your API wrapper, the methods you created in it, and why you decided to create those methods/how they were helpful | 
What was an edge case or failure case test you wrote for your API Wrapper? What was a nominal case? |
How does VCR aid in testing an API? |
What is the Heroku URL of your deployed application? | https://goeun-muncher.herokuapp.com/

### CODE REVIEW

File: lib/edamam_api_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        @recipe_list << create_recipe(recipe_data)
+      end
+    end
+    return @recipe_list
+  end
+
+  def self.find_recipe(params)
+    @recipe_list.each do |recipe|
+      if recipe.name == params
+        return recipe
```
Comment (droberts-sea) [inline lib/edamam_api_wrapper.rb:26]: This only works if the recipe being shown is in the most recent search results. If you open up two browser tabs and search for different things, you can only click on results from the last search.

The right thing to do here would be to send another request to Edamam, to get the details for that recipe. To do so you probably need something more distinct than just the recipe name - the URI might work nicely.

In general you should avoid using instance variables in class methods - these are basically global variables, and should only be used in very specific circumstances.

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Rails.application.routes.draw do
+
+  root 'recipes#search'
+  get 'index', to: 'recipes#index', as: 'recipes_index'
+  get ':recipe/show', to: 'recipes#show', as: 'recipes_show'
```
Comment (droberts-sea) [inline config/routes.rb:5]: These paths aren't quit RESTful. Why not use `resources [:index, :show]` here?

File: app/views/recipes/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    <div class="name"><h2><%= recipe.name %></h2></div>
+    <img src=<%= recipe.image %> class="thumbnail_photo">
+    <div class="info_link">
+      <%=link_to "More Info", recipes_show_path(recipe.name) %>
+    </div>
+  </li>
+  <% end %>
+  <% end %>
+</ul>
+</class>
```
Comment (droberts-sea) [inline app/views/recipes/index.html.erb:19]: What is this `</class>` tag?

**Review by droberts-sea — COMMENTED** [review]
We talked about this in our 1-on-1 today. Full review to come later in the week, once you've gotten a chance to update this.

File: app/views/recipes/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+<% if @recipes %>
+<ul class="index_container">
+  <% if @recipes == [] %>
+  "There are no munchies corresponding to the search!"
+  <% else %>
+  <% @recipes.each do |recipe| %>
+  <li class="recipe_card">
+    <div class="name"><h2><%= recipe.name %></h2></div>
+    <img src=<%= recipe.image %> class="thumbnail_photo">
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:11]: Suggestion:  Use `image_tag` and make the image a hyperlink.  It's human nature to want to click on the link.

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** |
Git hygiene |  Not very many commits, but good commit messages
Comprehension questions |  You only answered the Heroku link, and how you explored the API.
**General** |
Rails fundamentals (RESTful routing, use of named paths) |  Check
Semantic HTML |  Overall pretty good, but just a note that you **can** use other container tags like `section`, `main`, `article` etc with Bootstrap styling class names.  You don't have to always use `div`.  
Errors are reported to the user |  An invalid recipe name will result in the app crashing.  
API Wrapper to handle the API requests |  Check, but see Dan's note on the `find_recipe` method.  The way you have it won't work with two browser tabs, or if the user bookmarks a specific recipe.  Instead you should make another API call with the recipe's `uri` field.  
Controller testing |  **Missing**
Lib testing |  **MISSING**
Search Functionality |  Check
List Functionality |  Check
Show individual item functionality |  Check, but it won't always work.
**Styling** |
List view shows 10 items at a time and/or has pagination |  **No Pagination**
Attractive and usable UI |  Simple, but nice looking, good responsiveness
**API Features** |
The App attributes Edamam |  Check
The VCR cassettes do not contain the API key | **NO TESTING**
**External Resources** |
Link to deployed app on Heroku |  Check
**Overall** |  Well you got the API to work, but you **didn't** test anything.  Clearly one thing to review is how to test an app that uses an API and how to use VCR in testing.  See my and Dan's comments.
